### DISCUSSION TITLE
Self hosting with config.js file and a renovate.json file in the folder scan

### DISCUSSION THREAD

**Discussion by nbeny** [renovatebot/renovate#9050] [Request Help] [answered]
Hello,

I'm trying to run renovate in self hosting with a configuration file. All is working with this one. If In a repository scan I add a renovate.json file it will not be read.

Here my self hosting configuration read:

```
module.exports = {
    // witch platform ?
    platform: "gitlab",

    // url of gitlab
    endpoint: process.env.GITLAB_API_URL,

    // ignore some dependencies
    ignoreDeps: [
        "google.golang.org/grpc",
        "python",
    ],

    // path to run the bot
    repositories: [
        "repo/p1",
        "repo/p2"
    ],

    extends: ["config:base"],

    // branch to run the renovate bot
    baseBranches: ["dev"],

    // rebase if multiple run are done on the same repository
    rebaseWhen: "auto",

    // create a label
    labels: ["deps"],

    // create related merge request in issue and create issue
    dependencyDashboard: true,

    packageRules: [
        {
            matchPaths: ["renovate.json"],
            labels: ["special-deps"],
        },
    ],
};
```
And I try to overwrite the config via the file `renovate.json` at the base project folder with it inside:

```
{
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "ignoreDeps": [
        "github.com/docker/docker",
        "alpine"
    ]
}
```
with It the ignored deps are only: google.golang.org/grpc, python. but do not add on the specific repo this one: github.com/docker/docker, alpine

Like that I could have ignored some deps for every project or for specific project.

It is possible ? If not like that, what is the good way to do that ?

**Comment by viceice** [comment, marked as answer]
is the `renovate.json` in repo root on default branch?  only that file is used by renovate, not the baseBranches.

**Reply by nbeny** [reply]
only on the master so not the dev branch like specified in the config ?

renovate.json is in the dev branch only for the moment

Thanks for you're answer !

**Reply by viceice** [reply]
yes, it's renovate only uses the repository default branch, which can be changed in repo settings of cause
